### ISSUE TITLE
[@rollup/plugin-node-resolve] Browser overrides not being applied to dependent packages.

### ISSUE THREAD

**Issue by keean** [rollup/plugins#1308]
- Rollup Plugin Name: plugin-node-resolve
- Rollup Plugin Version: 14.1.0
- Rollup Version: 2.79.1
- Operating System (or Browser): Linux 
- Node Version: v16.17.0
- Link to reproduction _(⚠️ read below)_: https://replit.com/@KeeanSchupke/rollup-issue#rollup.config.js

### Expected Behavior

The AWS-SDK contains the following browser overrides in the package.json referenced as modules in the import in 'src/test.ts':

node_modules/@aws-sdk/client-chime-sdk-messaging/package.json:
```
"browser": {
    "./dist-es/runtimeConfig": "./dist-es/runtimeConfig.browser"
},
```

I have included 'rollup-plugin-sizes' with `{details: true}` to see which files are included in the bundle. The rollup-config.js file contains:
```
nodeResolve({
    mainFields: ['browser'],
}),
```

If we run from the shell like this:

```
> npm install
> npx rollup -c >& temp
> grep -a runtimeConfig temp
```

We expect to see:
```
dist-cjs/runtimeConfig.browser.js - 4.93 KB (1.15%)
    dist-cjs/runtimeConfig.shared.js - 1.1 KB (0.26%)
    dist-cjs/runtimeConfig.browser.js - 4.96 KB (5.15%)
    dist-cjs/runtimeConfig.shared.js - 1.08 KB (1.12%)
    dist-cjs/runtimeConfig.browser.js - 4.54 KB (7.93%)
    dist-cjs/runtimeConfig.shared.js - 1.08 KB (1.89%)
    /home/runner/rollup-issue/node_modules/@aws-sdk/client-sso/dist-cjs/runtimeConfig.shared.js?commonjs-exports - 73 B (0.13%)
    /home/runner/rollup-issue/node_modules/@aws-sdk/client-sts/dist-cjs/runtimeConfig.shared.js?commonjs-exports - 73 B (0.13%)
    /home/runner/rollup-issue/node_modules/@aws-sdk/client-chime-sdk-messaging/dist-cjs/runtimeConfig.shared.js?commonjs-exports - 73 B (0.13%)
    /home/runner/rollup-issue/node_modules/@aws-sdk/client-chime-sdk-messaging/dist-cjs/runtimeConfig.browser.js?commonjs-exports - 59 B (0.11%)
    /home/runner/rollup-issue/node_modules/@aws-sdk/client-sts/dist-cjs/runtimeConfig.browser.js?commonjs-exports - 59 B (0.11%)
    /home/runner/rollup-issue/node_modules/@aws-sdk/client-sso/dist-cjs/runtimeConfig.browser.js?commonjs-exports - 59 B (0.11%)
```

### Actual Behavior

What we actually see is:
```
dist-cjs/runtimeConfig.js - 4.93 KB (1.15%)
    dist-cjs/runtimeConfig.shared.js - 1.1 KB (0.26%)
    dist-cjs/runtimeConfig.js - 4.96 KB (5.15%)
    dist-cjs/runtimeConfig.shared.js - 1.08 KB (1.12%)
    dist-cjs/runtimeConfig.js - 4.54 KB (7.93%)
    dist-cjs/runtimeConfig.shared.js - 1.08 KB (1.89%)
    /home/runner/rollup-issue/node_modules/@aws-sdk/client-sso/dist-cjs/runtimeConfig.shared.js?commonjs-exports - 73 B (0.13%)
    /home/runner/rollup-issue/node_modules/@aws-sdk/client-sts/dist-cjs/runtimeConfig.shared.js?commonjs-exports - 73 B (0.13%)
    /home/runner/rollup-issue/node_modules/@aws-sdk/client-chime-sdk-messaging/dist-cjs/runtimeConfig.shared.js?commonjs-exports - 73 B (0.13%)
    /home/runner/rollup-issue/node_modules/@aws-sdk/client-chime-sdk-messaging/dist-cjs/runtimeConfig.js?commonjs-exports - 59 B (0.11%)
    /home/runner/rollup-issue/node_modules/@aws-sdk/client-sts/dist-cjs/runtimeConfig.js?commonjs-exports - 59 B (0.11%)
    /home/runner/rollup-issue/node_modules/@aws-sdk/client-sso/dist-cjs/runtimeConfig.js?commonjs-exports - 59 B (0.11%)
```

So the 'browser' override in the package.json of the aws-sdk is being ignored, and bundle produced by Rollup will not run in the browser.

**Comment by keean** [comment]
I just realised that its including the cjs version of the package. If i change the node resolve to:
```
nodeResolve({
    mainFields: ['module', 'browser'],
}),
```
Then run rollup:
```
npx rollup -c rollup-config.js >& temp
grep -a runtimeConfig temp
```
We get the following files bundled:
```
dist-es/runtimeConfig.js - 4.65 KB (0.88%)
dist-es/runtimeConfig.shared.js - 961 B (0.18%)
dist-es/runtimeConfig.js - 4.65 KB (4.39%)
dist-es/runtimeConfig.shared.js - 945 B (0.87%)
dist-es/runtimeConfig.js - 4.27 KB (7.28%)
dist-es/runtimeConfig.shared.js - 945 B (1.57%)
```
This clearly matches the 'browser' override, and we can see it is still not being applied.

**Comment by keean** [comment]
This is odd... the same versions of rollup and plugins on Replit gets the correct files in the bundle:
```
grep -a runtimeConfig temp
```
gives the following:
```
dist-es/runtimeConfig.browser.js - 2.58 KB (0.72%)
dist-es/runtimeConfig.shared.js - 465 B (0.13%)
```
Which looks correct, but I have no idea why, as I just copied the test case files across from my local machine to Replit.

**Comment by keean** [comment]
Looks like some update has already fixed this... deleting node_modules and package-lock.json and re-running `npm install` gets the correct output on my local machine, so something has changed that has fixed this.

**Comment by keean** [comment]
Okay my app build is still having the same problem, and I have tracked it down to the 'rollup-plugin-includepaths' plugin.

Without the plugin:
```
import nodeResolve from '@rollup/plugin-node-resolve';
import commonjs from '@rollup/plugin-commonjs';
import typescript from 'rollup-plugin-typescript2';
import json from '@rollup/plugin-json';
import sizes from 'rollup-plugin-sizes';
import includePaths from 'rollup-plugin-includepaths';

export default [
    {
        input: ['src/test.ts'],
        output: {
            format: 'iife',
            sourcemap: true,
            name: 'client',
            file: 'build/client.js'
        },
        plugins: [
            //includePaths(),
            nodeResolve({
                mainFields: ['module','browser'],
            }),
            commonjs(),
            json(),
            typescript(),
            sizes({details: true}),
        ],
    },
]
```
Running in the shell:
```
npx rollup -c >& temp
grep -a runtimeConfig temp
```
We get the correct output:
```
    dist-es/runtimeConfig.browser.js - 2.58 KB (0.72%)
    dist-es/runtimeConfig.shared.js - 465 B (0.13%)
```
Adding the plugin:
```
import nodeResolve from '@rollup/plugin-node-resolve';
import commonjs from '@rollup/plugin-commonjs';
import typescript from 'rollup-plugin-typescript2';
import json from '@rollup/plugin-json';
import sizes from 'rollup-plugin-sizes';
import includePaths from 'rollup-plugin-includepaths';

export default [
    {
        input: ['src/test.ts'],
        output: {
            format: 'iife',
            sourcemap: true,
            name: 'client',
            file: 'build/client.js'
        },
        plugins: [
            includePaths(),
            nodeResolve({
                mainFields: ['module','browser'],
            }),
            commonjs(),
            json(),
            typescript(),
            sizes({details: true}),
        ],
    },
]
```
Running in the shell:
```
npx rollup -c >& temp
grep -a runtimeConfig temp
```
We now get multiple copies of the wrong version:
```
    dist-es/runtimeConfig.js - 3.18 KB (0.89%)
    dist-es/runtimeConfig.shared.js - 465 B (0.13%)
    dist-es/runtimeConfig.js - 3.19 KB (4.74%)
    dist-es/runtimeConfig.shared.js - 449 B (0.65%)
    dist-es/runtimeConfig.js - 2.88 KB (7.52%)
    dist-es/runtimeConfig.shared.js - 449 B (1.14%)
```

**Comment by FranklinWhale** [comment]
Update rollup-plugin-typescript2 may help: https://github.com/rollup/plugins/issues/1267#issuecomment-1267981557

**Comment by keean** [comment]
@FranklinWhale I think that is what caused the original test case to start working mysteriously yesterday. It didn't work, and then after an 'npm install' it did work.

However, as I noted above it hasn't fixed my application, so I tracked the issue down to `includePaths` and I have updated https://replit.com/@KeeanSchupke/rollup-issue to show the use-case that is still failing.

It is interesting that with `includePaths` not only is the wrong `runtimeConfig` included in the bundle (ignoring the `browser` override) but the file is included multiple times.
